usaspending_api/search/tests/integration/test_spending_by_award.py

```diff
@@ -884,6 +887,100 @@ def _test_correct_response_for_psc_codes(client):
+            }
+        ),
+    )
+    expected_result = [{"internal_id": 1, "Award ID": "abc111", "generated_internal_id": "CONT_AWD_TESTING_1"}]
+    assert resp.status_code == status.HTTP_200_OK
+    assert len(resp.json().get("results")) == 1
+    assert resp.json().get("results") == expected_result, "PSC Code filter does not match expected result"
+
+
+def _test_correct_response_for_psc_code_list_subawards(client):
+    """ As of this writing, subawards query postgres whereas prime awards query elasticsearch.  Let's test both. """
+    resp = client.post(
+        "/api/v2/search/spending_by_award",
+        content_type="application/json",
+        data=json.dumps(
+            {
+                "filters": {
+                    "award_type_codes": ["A", "B", "C", "D"],
+                    "psc_codes": ["PSC2"],
+                    "time_period": [{"start_date": "2007-10-01", "end_date": "2020-09-30"}],
+                },
+                "fields": ["Sub-Award ID"],
+                "page": 1,
+                "limit": 60,
+                "sort": "Sub-Award ID",
+                "order": "desc",
+                "subawards": True,
+            }
+        ),
+    )
+    expected_result = [
+        {
+            "internal_id": "11111",
+            "prime_award_internal_id": 1,
+            "Sub-Award ID": "11111",
+            "prime_award_generated_internal_id": "CONT_AWD_TESTING_1",
+        }
+    ]
+    assert resp.status_code == status.HTTP_200_OK
+    assert len(resp.json().get("results")) == 1
+    assert resp.json().get("results") == expected_result, "PSC Code filter does not match expected result"
+
+
+def _test_correct_response_for_psc_code_object_subawards(client):
+    """ As of this writing, subawards query postgres whereas prime awards query elasticsearch.  Let's test both. """
+    resp = client.post(
+        "/api/v2/search/spending_by_award",
+        content_type="application/json",
+        data=json.dumps(
+            {
+                "filters": {
+                    "award_type_codes": ["A", "B", "C", "D"],
+                    "psc_codes": {"require": ["PSC2"], "exclude": ["PSC0"]},
+                    "time_period": [{"start_date": "2007-10-01", "end_date": "2020-09-30"}],
+                },
+                "fields": ["Sub-Award ID"],
+                "page": 1,
+                "limit": 60,
+                "sort": "Sub-Award ID",
+                "order": "desc",
+                "subawards": True,
+            }
+        ),
+    )
+    expected_result = [
+        {
+            "internal_id": "11111",
+            "prime_award_internal_id": 1,
+            "Sub-Award ID": "11111",
+            "prime_award_generated_internal_id": "CONT_AWD_TESTING_1",
+        }
+    ]
+    assert resp.status_code == status.HTTP_200_OK
+    assert len(resp.json().get("results")) == 1
+    assert resp.json().get("results") == expected_result, "PSC Code filter does not match expected result"
+
+
```

Comment:
There don't seem to be any tests here checking "eclipsing" logic (my term for where a path in one of the lists is a child of a member of the other list, see https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/search/tests/integration/hierarchical_filters/test_tas_filter_heirarchical_cases.py#L128 and down for some example cases)